src/olympia/abuse/views.py

```diff
@@ -167,34 +171,62 @@ def process_datestamp(date_string):
 def cinder_webhook(request):
-    if request.data.get('event') == 'decision.created' and (
-        payload := request.data.get('payload', {})
-    ):
+    try:
+        if request.data.get('event') != 'decision.created':
+            log.info('Non-decision payload received: %s', str(request.data)[:255])
+            raise ValidationError('Not a decision')
+
+        if not (payload := request.data.get('payload', {})):
+            log.info('No payload received: %s', str(request.data)[:255])
+            raise ValidationError('No payload')
+
         source = payload.get('source', {})
         job = source.get('job', {})
-        if (queue_name := job.get('queue', {}).get('slug')) == CinderEntity.queue:
-            log.info('Valid Payload from AMO queue: %s', payload)
-            job_id = job.get('id', '')
-            decision_id = source.get('decision', {}).get('id')
-            actions = payload.get('enforcement_actions')
-            try:
-                cinder_report = CinderReport.objects.get(job_id=job_id)
-                cinder_report.process_decision(
-                    decision_id=decision_id,
-                    decision_date=process_datestamp(payload.get('timestamp')),
-                    decision_actions=actions,
-                )
-            except CinderReport.DoesNotExist:
-                log.debug('CinderReport instance not found for job id %s', job_id)
-        else:
+        if (queue_name := job.get('queue', {}).get('slug')) != CinderEntity.queue:
             log.info('Payload from other queue: %s', queue_name)
-    else:
-        log.info(
-            'Non-decision payload received: %s',
-            str(request.data)[:255],
+            raise ValidationError('Not from a queue we process')
+
+        if (
+            not (actions := payload.get('enforcement_actions'))
+            or len(actions) > 1
+            or not CinderReport.DECISION_ACTIONS.has_api_value(actions[0])
+        ):
+            log.exception(
+                'Cinder webhook request failed: bad enforcement_actions [%s]', actions
+            )
+            raise ValidationError('Payload invalid: "decision_actions" malformed')
+
+        log.info('Valid Payload from AMO queue: %s', payload)
+        job_id = job.get('id', '')
+        try:
+            cinder_report = CinderReport.objects.get(job_id=job_id)
+        except CinderReport.DoesNotExist:
+            log.debug('CinderReport instance not found for job id %s', job_id)
+            raise ValidationError('No matching job id found')
+
+        cinder_report.process_decision(
+            decision_id=source.get('decision', {}).get('id'),
+            decision_date=process_datestamp(payload.get('timestamp')),
+            decision_action=CinderReport.DECISION_ACTIONS.for_api_value(
+                actions[0]
+            ).value,
         )
 
-    return Response(data={'amo-received': True}, status=status.HTTP_201_CREATED)
+    except ValidationError as exc:
```

Comment:
mmm, I would have expected us to keep returning 400s for errors in requests meant for us, and just return 200s when it's for another queue.

Reply:
> If the receiving server returns an error response, Cinder will continue retrying to send that webhook indefinitely (with expotential backoff). That way, even if your endpoint is down, the webhooks will come in once you're back online. Cinder expects a server response within 10 seconds, otherwise the request will timeout and be retried.
>
>HTTP response codes between 400 and 599 are all considered errors.

I'm reading that as if we return an error code cinder will just retry again which doesn't seem helpful.

Reply:
Ugh ok. Could you add a comment in that `except` block explaining that (maybe more succinctly) ?